### DISCUSSION TITLE
Is it possible to grab a field of a model and its parent to make it more clear which instance in a file is the problem

### DISCUSSION THREAD

**Discussion by matroscoe** [pydantic/pydantic#5061] [Question]
I have a model which is made up of a list of other models and the source we are pulling from can provide varying lengths of lists. I am wondering if there is a way to pull a field from both the model failing validation and possibly its parent to provide a more helpful error message so our users can search on that instead of the number that is currently provided which gives the index in the list. I am not looking to have the library do this for me automatically as how would it know which field to use I just need help figuring out how to tweak the error message to be more useful to our users. 

e.g. instead of 
``` python
parent -> 4 -> current_model -> 9 -> value
  field required (type=value_error.missing)
```

The error message would be more like: 
``` python
parent -> Parent(name="<parent.name>") -> current_model -> CurrentModel(name="current_model.name") -> value
  field required (type=value_error.missing)
```

**Comment by dmontagu** [comment]
```python
from pydantic import ValidationError

try:
    <code that may raise a ValidationError>
except ValidationError as exc:
    errors = exc_info.errors()
    <code that does something with errors>
```
The `exc_info.errors()` should be a list of objects that contains the data that is used to produce the error message. If you want to do something more interesting with that data, it should be possible through code like this.

Does that get you what you need?

**Reply by matroscoe** [reply]
kind of, I was hoping I could catch the exception text when it was initialized or put this on a base class so that it would have this exception formatting across the board (but just in this one package). You solution would definitely work I am just hoping there is a more "elegant" way to do this than having this try/except repeated in a lot of different models.

**Reply by dmontagu** [reply]
I understand now, I think this is a reasonable request but I'm not sure there is a good way to achieve it.

And even if we were able to make it work in v1, it would need to be completely written for v2 due to how we've moved validation logic into pydantic_core and stripped out some of the context you'd need (in exchange for massive performance gains).

In the short term I think the vast majority of our focus will be on getting v2 released with just the functionality already planned, so I don't think we'll be able to get to this soon, but if you want to create a feature request you can open an issue.

@samuelcolvin just making sure -- you don't see an easy way to manipulate the errors produced to include information about the "source model" that a field came from do you? In principle it feels like maybe we (or users.. :) could include information like that in the typed_dict_schema metadata and use it when formatting the errors, but I don't know how involved it would be, or if it's even possible given that's possibly happening in rust code.

**Reply by matroscoe** [reply]
No problem I totally understand if this would end up being something we have to wait for v2 and then how to implement it there. Once you determine how I should open the feature request if it is confirmed new functionality let me know and I will happily open the ticket.
